packages/utils/node-resolver-rs/src/specifier.rs

```diff
@@ -41,22 +40,22 @@ where
 }
 
 #[derive(PartialEq, Eq, Hash, Clone, Debug)]
-pub enum Specifier<'a> {
-  Relative(Cow<'a, Path>),
-  Absolute(Cow<'a, Path>),
-  Tilde(Cow<'a, Path>),
-  Hash(Cow<'a, str>),
-  Package(Cow<'a, str>, Cow<'a, str>),
-  Builtin(Cow<'a, str>),
-  Url(&'a str),
+pub enum Specifier {
+  Relative(PathBuf),
+  Absolute(PathBuf),
+  Tilde(PathBuf),
+  Hash(String),
+  Package(String, String),
+  Builtin(String),
+  Url(String),
 }
```

Comment:
This also seems quite unrelated from the cache changes.

Reply:
This is simplifying the code.

I understand `Cow` is here as a performance optimisation, but it doesn't seem like this makes a difference to performance.

Reply:
That seems unrelated to the safety issues though. Small things like not copying specifiers do add up in my experience. Perhaps it doesn't appear to make a difference right now because the rest of your build is a larger portion of the overall time than the resolver.

Reply:
Would you know +/- what types of benchmarks you've ran? I would be keen to add some benchmarks we can use to measure this.

Note that on the benchmarks I've posted above, there is between 1-10% difference of the overall time. However, since this is quite fast as is I don't that will translate to a meaningful difference in build time. I'm quite confident we can optimise this much more by optimising IO and JSON parsing.

I've added a few notes on the benchmark suite I added.

Reply:
I'd need to run them again, it was during development several years ago. Just seems strange to remove optimizations that we already implemented that aren't causing any particular problem.

Reply:
I'm asking what you've benchmarked so we can understand and reproduce your findings.

Regardless of whether you run them again, we needed to know what types of benchmarks you're running and what the results are.

- - -

We won't remove any optimizations. We are only trying to measure things.

Reply:
Usually end-to-end benchmarks like esbuild or https://github.com/web-infra-dev/bundler-benchmark